tests/smoke/support/page/files/sharedWithMe.ts

```diff
@@ -11,4 +12,24 @@ export class SharedWithMePage {
     const { page } = this.actor
     await page.click('a[href="#/files/list/shared-with-me"]')
   }
+
+  async acceptShares({ name }: { name: string }): Promise<void> {
+    const { page } = this.actor
+    const startUrl = page.url()
+
+    this.navigate()
+    const objectExists = await cta.files.resourceExists({
+      page: page,
+      name: name
+    })
+
+    const statusResource = await page.textContent(
+      `//*[@data-test-resource-name="${name}"]/../../../../../td[3]/div[1]/span[1]`
```

Comment:
the xpath "/../../../../../" looks a bit fragile to me, is there a better way of selecting the span?

Reply:
Yes, I agree with you. I fixed it ``//*[@data-test-resource-name="${name}"]/ancestor::tr//span[contains(@class, "file-row-share-status-text")]``